src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -696,16 +693,48 @@ private IOperation CreateBoundObjectInitializerMemberOperation(BoundObjectInitia
             }
         }
 
-        private ICollectionElementInitializerOperation CreateBoundCollectionElementInitializerOperation(BoundCollectionElementInitializer boundCollectionElementInitializer)
+        private IOperation CreateBoundCollectionElementInitializerOperation(BoundCollectionElementInitializer boundCollectionElementInitializer)
         {
-            IMethodSymbol addMethod = boundCollectionElementInitializer.AddMethod;
-            Lazy<ImmutableArray<IOperation>> arguments = new Lazy<ImmutableArray<IOperation>>(() => boundCollectionElementInitializer.Arguments.SelectAsArray(n => Create(n)));
-            bool isDynamic = false;
+            MethodSymbol addMethod = boundCollectionElementInitializer.AddMethod;
             SyntaxNode syntax = boundCollectionElementInitializer.Syntax;
             ITypeSymbol type = boundCollectionElementInitializer.Type;
             Optional<object> constantValue = ConvertToOptional(boundCollectionElementInitializer.ConstantValue);
             bool isImplicit = boundCollectionElementInitializer.WasCompilerGenerated;
-            return new LazyCollectionElementInitializerExpression(addMethod, isDynamic, arguments, _semanticModel, syntax, type, constantValue, isImplicit);
+
+            if (IsMethodInvalid(boundCollectionElementInitializer.ResultKind, addMethod))
+            {
+                return CreateInvalidExpressionForHasArgumentsExpression(boundCollectionElementInitializer.ImplicitReceiverOpt, boundCollectionElementInitializer.Arguments, null, syntax, type, constantValue, isImplicit);
+            }
+
+            Lazy<IOperation> receiver;
+            if (boundCollectionElementInitializer.InvokedAsExtensionMethod)
+            {
+                Debug.Assert(boundCollectionElementInitializer.Arguments.Length >= 1);
+                Debug.Assert(boundCollectionElementInitializer.Arguments[0].Kind == BoundKind.ImplicitReceiver);
+                Debug.Assert(boundCollectionElementInitializer.ImplicitReceiverOpt == null);
+                receiver = OperationFactory.NullOperation;
```

Comment:
>receiver = OperationFactory.NullOperation; [](start = 16, length = 42)

I am not sure why do we have this special case here, it looks like CreateReceiverOperation is able to handle this case just fine for the BoundCall node.